core/embed/rust/src/ui/model_tr/component/title.rs

```diff
@@ -0,0 +1,120 @@
+use crate::{
+    micropython::buffer::StrBuffer,
+    time::Instant,
+    ui::{
+        component::{Component, Event, EventCtx, Marquee, Never},
+        display,
+        geometry::{Offset, Rect},
+    },
+};
+
+use super::super::theme;
+
+pub struct Title {
+    area: Rect,
+    title: StrBuffer,
+    marquee: Marquee<StrBuffer>,
+    needs_marquee: bool,
+    centered: bool,
+}
+
+impl Title {
+    pub fn new(title: StrBuffer) -> Self {
+        Self {
+            title: title.clone(),
+            marquee: Marquee::new(title, theme::FONT_HEADER, theme::FG, theme::BG),
+            needs_marquee: false,
+            area: Rect::zero(),
+            centered: false,
+        }
+    }
+
+    pub fn with_centered(mut self) -> Self {
+        self.centered = true;
+        self
+    }
+
+    pub fn get_text(&self) -> &str {
+        self.title.as_ref()
+    }
+
+    pub fn set_text(&mut self, ctx: &mut EventCtx, new_text: StrBuffer) {
+        self.title = new_text.clone();
+        self.marquee.set_text(self.title.clone());
+        let text_width = theme::FONT_HEADER.text_width(new_text.as_ref());
+        self.needs_marquee = text_width > self.area.width();
```

Comment:
It could be done, but then the `marquee` would essentially be the `title` itself - I prefer having it separated like this - so `title` can work even without `marquee`